pythonAPIClient/client.py

```diff
@@ -297,9 +297,159 @@ def usage_by_client(self, clientID, owner):
+                    Url to retrieve stream from the camera.
+
+                Raises
+                ------
+                    AuthenticationError
+                        If the client secret of this client object does not match the clientID.
+                    FormatError
+                        List of invalid attributes.
+                    ResourceConflictError
+                        The legacy_cameraID already exist in the database.
+                    InternalError
+                        If there is an API internal error.
+                    ResourceNotFoundError
+                        If no client app exists with the clientID of this client object.
+
+                Returns
+                -------
+                str
+                    The new camera ID for the successfully added camera.
+        """
+
+        local_params = dict(locals())
+
+        if self.token is None:
+            self.request_token()
+
+        url = Client.base_URL + 'cameras/create'
+
+        local_params['type'] = local_params.pop('camera_type')
+        del local_params['self']
+
+        if camera_type == 'ip':
+            if ip is None:
+                response = self._check_token(requests.post(url, data=local_params,
+                                                           headers=self.header_builder()),
+                                             flag='POST', url=url, data=local_params)
+                raise FormatError('Must provide ip')
+            local_params['retrieval'] = {
+                'ip': local_params.pop('ip'),
+                'port': local_params.pop('port'),
+                'brand': local_params.pop('brand'),
+                'model': local_params.pop('model'),
+                'image_path': local_params.pop('image_path'),
+                'video_path': local_params.pop('video_path')
+            }
+        elif camera_type == 'non-ip':
+            if snapshot_url is None:
+                response = self._check_token(requests.post(url, data=local_params,
+                                                           headers=self.header_builder()),
+                                             flag='POST', url=url, data=local_params)
+                raise FormatError('Must provide snapshot_url')
+
+            local_params['retrieval'] = {
+                'snapshot_url': local_params.pop('snapshot_url')
+            }
+        elif camera_type == 'stream':
+            if m3u8_url is None:
+                response = self._check_token(requests.post(url, data=local_params,
+                                                           headers=self.header_builder()),
+                                             flag='POST', url=url, data=local_params)
+                raise FormatError('Must provide m3u8_url')
+            local_params['retrieval'] = {
+                'm3u8_url': local_params.pop('m3u8_url')
+            }
+        else:
+            raise FormatError('Must provide a legal camera_type')
```

Comment:
and we do not need to raise error here. say a user did not provide 'ip' when the camera_type is ip, and we did not add a 'retrieval' to dictionary ( or set it to None), the API will return the error when we call it with all parameters